x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/utils/persistence/ResultsPersisterService.java

```diff
@@ -42,6 +44,22 @@
 import java.util.stream.Collectors;
 
 public class ResultsPersisterService {
+    /**
+     * List of rest statuses that we consider irrecoverable
+     */
+    public static final Set<RestStatus> IRRECOVERABLE_REST_STATUSES = new HashSet<>(
```

Comment:
nit: since the name is in capitals it should be immutable, i.e. `Set.of()` or `Collections.unmodifiableSet()`.